src/service_manager.rs

```diff
@@ -208,4 +209,41 @@ impl ServiceManager {
             Ok(Service::new(unsafe { ScHandle::new(service_handle) }))
         }
     }
+
+    /// Retrieves the service name of the specified service.
+    pub fn get_service_key_name(&self, display_name: impl AsRef<OsStr>) -> Result<OsString> {
+        let service_display_name =
+            WideCString::from_os_str(display_name).map_err(|_| Error::DisplayNameHasNulByte)?;
+        let mut size = 0u32;
+        let res = unsafe {
+            Services::GetServiceKeyNameW(
+                self.manager_handle.raw_handle(),
+                service_display_name.as_ptr(),
+                std::ptr::null_mut(),
+                &mut size as _,
+            )
+        };
+
+        if res == 0 && unsafe { GetLastError() } != ERROR_INSUFFICIENT_BUFFER {
+            return Err(Error::Winapi(io::Error::last_os_error()));
+        }
+
+        let mut buffer = vec![0u16; size as usize];
+        let res = unsafe {
+            Services::GetServiceKeyNameW(
+                self.manager_handle.raw_handle(),
+                service_display_name.as_ptr(),
+                buffer.as_mut_ptr(),
+                &mut size as _,
+            )
+        };
+
+        if res == 0 {
```

Comment:
Nitpick. But please call the variable `result`. Even if it's a commonly used abbreviation it does not really serve any purpose other than making the code harder to read. Adding `ult` to the variable name will not split any code in more lines or anything.

Reply:
Done